aot/src/main/java/com/dylibso/chicory/aot/AotMachine.java

```diff
@@ -688,10 +692,10 @@ private void compileBody(
                     break;
                 case BLOCK:
                 case LOOP:
-                    ctx.enterScope(ins.scope(), blockType(ins));
+                    ctx.enterScope(ins.scope().get(), blockType((int) ins.operands()[0]));
```

Comment:
Using `Optional` for these fields hasn't improved type safety much, as the caller is unconditionally getting the values. The advantage is that it will fail immediately, rather than providing a null. What we don't have is enforcement between the opcode type and the type-specific fields.

We could improve this by making these methods return the values directly, but throw `IllegalStateException`  if the opcode type is not valid. At construction time, we can enforce that the metadata values are set if and only if the opcode type matches.

In other words, `scope()` must be set for, and can only be set for and called for, the `BLOCK`, `LOOP`, `END`, and `IF` types. Similar logic applies to the other fields.

Reply:
done